### PR TITLE
refactor: Update storage entities

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-596

All tests will pass in the final [PR](https://github.com/keboola/keboola-as-code/pull/1770).

[Large set of fixes and E2e tests](https://github.com/keboola/keboola-as-code/pull/1759) is divided into smaller PRs to do a review.

--------------

**Changes:**
- Updated entities from the `storage` domain.

-----------

Commend per commits.

### CODE REVIEW

File: internal/pkg/service/stream/storage/model/retry.go [hunk 1]
```diff
@@ -75,7 +75,7 @@ func (b *retryBackoff) RetryAt(failedAt time.Time, attempt int) (retryAt time.Ti
 	return failedAt.Add(time.Duration(float64(interval) * randomFactor))
 }
 
-func (v *Retryable) IncrementRetry(backoff RetryBackoff, failedAt time.Time, reason string) {
+func (v *Retryable) IncrementRetryAttempt(backoff RetryBackoff, failedAt time.Time, reason string) {
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/model/retry.go:78]: Improved naming.

File: internal/pkg/service/stream/storage/model/file.go [hunk 2]
```diff
@@ -17,14 +17,15 @@ import (
...
-	FileTypeCSV = "csv"
+	FileTypeCSV = FileType("csv")
 )
 
 // File represents a file prepared in the staging storage to be imported into the target storage.
 // File consists from zero or more Slices.
 type File struct {
 	FileKey
 	Retryable
+	Deleted        bool                  `json:"-"` // internal field to mark the entity for deletion, there is no soft delete
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/model/file.go:28]: File/Slice entity have new `Deleted` field, so in the `plugin.Collection.OnFile/SliceSave` we can detect the entity deletion, and (for example) delete/rollup also its statistics.

Entities in the storage repository have no soft-delete, they aren't definitions.

File: internal/pkg/service/stream/storage/level/local/slice.go [hunk 3]
```diff
@@ -25,18 +25,17 @@ type Slice struct {
...
 	if err != nil {
 		return Slice{}, err
 	}
 
 	return Slice{
-		Dir:                filepath.Join(f.Dir, NormalizeDirPath(path)),
-		Filename:           filename,
-		Compression:        f.Compression,
-		DiskSync:           f.DiskSync,
-		AllocatedDiskSpace: f.DiskAllocation.ForNextSlice(prevSliceSize),
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/local/slice.go:40]: `AllocatedDiskSpace` field will be set by the `statistics` domain, to keep in the separation of individual domains and the they connection via the `plugin` pkg.

File: internal/pkg/service/stream/storage/level/staging/file.go [hunk 4]
```diff
@@ -1,28 +1,34 @@
...
+	// Provider of the staging file.
+	Provider FileProvider `json:"provider" validate:"required"`
 	// Compression configuration.
-	Compression compression.Config `json:"compression"  `
-	// UploadCredentials to the staging storage.
-	UploadCredentials           *keboola.FileUploadCredentials `json:"credentials" validate:"required"`
-	UploadCredentialsExpiration utctime.UTCTime                `json:"credentialsExpiration" validate:"required"`
+	Compression compression.Config `json:"compression" validate:"required"`
+	// Expiration determines how long it is possible to write to the staging file, e.g. due to expiring credentials.
+	Expiration utctime.UTCTime `json:"expiration" validate:"required"`
```
Comment (michaljurecko) [inline internal/pkg/service/stream/storage/level/staging/file.go:23]: File/Slice entity is now more general, I removed Keboola specific fields, and added `Provider` field.

The upload credentials are managed by the `keboolasink.Bridge` and have own etcd prefix.

**Review by Matovidlo — APPROVED** [review]
LGTM

**Review by hosekpeter — APPROVED** [review]
LGTM

**Pull request merged**